### PR TITLE
Enable some stricter compiler and linting options, and general repo cleanup.

### PR DESCRIPTION
**Pull request by iclanton**

### CODE REVIEW

**Review by D4N14L — APPROVED** [review]

File: .gitattributes [hunk 1]
```diff
@@ -64,7 +64,7 @@
 
 # Don't allow people to merge changes to these generated files, because the result
 # may be invalid.  You need to run "rush update" again.
-pnpm-lock.yaml               merge=binary
+pnpm-lock.yaml               merge=text
```
Comment (octogonz) [inline .gitattributes:67]: This is generated from the `rush init` template.  If the best practice has changed, we should reflect that in the template

Comment (octogonz) [reply .gitattributes:67]: Also this new line is inconsistent with the `#` comment above it

File: .gitignore [hunk 2]
```diff
@@ -31,8 +31,8 @@ bower_components
 build/Release
 
 # Dependency directories
-node_modules/
-jspm_packages/
+node_modules
+jspm_packages
```
Comment (octogonz) [inline .gitignore:35]: The pattern `node_modules` matches either a file or folder, whereas `node_modules/` matches a folder.

As a general convention, this practice therefore seems wrong:
1. It accidentally matches files which is slightly less efficient
2. For human readers of this file, it no longer clearly communicates that we expect `node_modules` to be a folder not a file

File: .gitignore [hunk 3]
```diff
@@ -58,11 +58,21 @@ jspm_packages/
 # OS X temporary files
 .DS_Store
 
+# IntelliJ IDEA project files; if you want to commit IntelliJ settings, this recipe may be helpful:
+# https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore
+.idea/
+*.iml
```
Comment (octogonz) [inline .gitignore:64]: This should be added to the `rush init` template

File: .gitignore [hunk 4]
```diff
@@ -58,11 +58,21 @@ jspm_packages/
...
 # Rush temporary files
 common/deploy/
 common/temp/
 common/autoinstallers/*/.npmrc
 **/.rush/temp/
+*.lock
+
+# Heft temporary files
+.cache
+.heft
```
Comment (octogonz) [inline .gitignore:75]: This overlaps with line 91 below

File: .gitattributes (hunk 1 above)
Comment (octogonz) [reply .gitattributes:67]: 😂 Didn't realize this was the TSDoc repo -- I thought it was the Rush Stack repo.

File: .gitignore (hunk 3 above)
Comment (octogonz) [reply .gitignore:64]: 😂 Didn't realize this was the TSDoc repo -- I thought it was the Rush Stack repo.

File: common/changes/@microsoft/tsdoc/general-updates_2024-05-28-08-28.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/tsdoc",
+      "comment": "Eliminate remaining const enums.",
```
Comment (octogonz) [inline common/changes/@microsoft/tsdoc/general-updates_2024-05-28-08-28.json:5]: Is this a major breaking change?

Comment (iclanton) [reply common/changes/@microsoft/tsdoc/general-updates_2024-05-28-08-28.json:5]: No, existing callers won't be broken.

File: .gitignore (hunk 3 above)
Comment (iclanton) [reply .gitignore:64]: This is from the `rush init` template :)

File: .gitattributes (hunk 1 above)
Comment (iclanton) [reply .gitattributes:67]: This is from `rush init`.

File: .gitignore (hunk 2 above)
Comment (iclanton) [reply .gitignore:35]: This comes from the `rush init` template, so we should fix it there.

**Pull request merged**